unstructured/partition/xlsx.py

```diff
@@ -201,44 +202,26 @@ def _get_connected_components(
         overlapping components to return distinct components.
     """
     max_row, max_col = sheet.shape
-    visited = set()
+    graph: nx.Graph = nx.grid_2d_graph(max_row, max_col)
+    node_array = np.indices((max_row, max_col)).T
+    empty_cells = sheet.isna().T
+    nodes_to_remove = [tuple(pair) for pair in node_array[empty_cells]]
+    graph.remove_nodes_from(nodes_to_remove)
+    connected_components_as_nodes = list(nx.connected_components(graph))
```

Comment:
do we need to convert it into list? seems we are just iterating over this once. So maybe keep it as generator to save some memory?

Reply:
Dropped the conversion to list in [1f43edd](https://github.com/Unstructured-IO/unstructured/pull/1761/commits/1f43eddf680cd5ee08b8b347ed93d4a5fc03dd36).